tinygrad/renderer/cstyle.py

```diff
@@ -41,7 +41,7 @@ def render_cast(self, x:str, var_dtype:DType, bitcast=False) -> str:
   def render_vectorize(self, x:List[str], var_dtype:DType) -> str:
     assert len(x) == var_dtype.count, f"cast is wrong size {len(x)} != {var_dtype.count}"
     assert self.float4 is not None, "vectorized cast is not supported on this platform"
-    return f"{self.float4.replace('float4', self.render_dtype(var_dtype))}({','.join(x)})"
+    return f"{self.float4.replace('float4', self.render_dtype(var_dtype))}" + (f"{{{','.join(x)}}}" if self.device == "CLANG" else f"({','.join(x)})")
```

Comment:
Why is clang different here?

Reply:
The custom vector type support the following direct initializations that don't work if used with parenthesis.

```
typedef float float4 __attribute__((aligned(16),vector_size(16)));
float4 x = (float4){0.0f,0.0f,0.0f,0.0f};
float4 y = (float4){0.0f};
```

For parenthesis to work, an initialization function is required, much like make_float4, but with no other benefit than that.

```
float4 make_float4(float a, float b, float c, float d) {
    return (float4){a, b, c, d};
}
float4 m = make_float4(1.0f, 2.0f, 3.0f, 4.0f);
```

Other backends like CUDA and AMD do use the initialization function, that's the reason for the clause there